src/main/groovy/lang/ObjectRange.java

```diff
@@ -37,7 +37,7 @@
  * Note: This class is similar to {@link IntRange}. If you make any changes to this
  * class, you might consider making parallel changes to {@link IntRange}.
  */
-public class ObjectRange extends AbstractList implements Range {
+public class ObjectRange extends AbstractList<Comparable> implements Range<Comparable> {
```

Comment:
Done, though I couldn't change every `Comparable` to `T`, because of the constructor (changing char to int and float to double).